lib/actions/EventLambdaSchedule.js

```diff
@@ -45,73 +45,95 @@ module.exports = function (S) {
+            Action: populatedEvent.config.enabled ? 'lambda:InvokeFunction' : 'lambda:DisableInvokeFunction',
             Principal: 'events.amazonaws.com',
-            Qualifier: _this.evt.options.stage
+            SourceArn: 'arn:aws:events:' + region + ':' + awsAccountId + ':rule/' + ruleName,
+            Qualifier: stage
           };
-          return _this.aws.request('Lambda', 'addPermission', params, _this.evt.options.stage, _this.evt.options.region)
+          return _this.aws.request('Lambda', 'addPermission', params, stage, region)
+            .then(function() {
+               SUtils.sDebug(`${stage} - ${region} - ${ruleName}: added lambda access policy statement`);
+            });
         })
-        .then(function (data) {
-
-          SUtils.sDebug(`Setting lambda ${lambdaArn}:${stage} as target for rule ${ruleName} for lambda ${functionName}`);
-
-          let params = {
-            Rule: ruleName,
-            Targets: [
-              {
-                Arn: lambdaArn + ":" + stage,
-                Id: functionName
-              }
-            ]
-          };
-
-          if (populatedEvent.config.input) {
-            params.Targets[0].Input = JSON.stringify(populatedEvent.config.input);
+        .then(function () {
+
+          if (populatedEvent.config.enabled) {
+            SUtils.sDebug(`Setting lambda ${lambdaArn}:${stage} as target for rule ${ruleName} for lambda ${functionName}`);
+
+            let params = {
+              Rule: ruleName,
+              Targets: [
+                {
+                  Arn: lambdaArn + ':' + stage,
+                  Id: functionName + '_' + stage,
+                  InputPath: _this.evt.options.inputPath
+                }
+              ]
+            };
+
+            if (populatedEvent.config.input) {
+              params.Targets[0].Input = JSON.stringify(populatedEvent.config.input);
+            }
+
+            return _this.aws.request('CloudWatchEvents', 'putTargets', params, stage, region)
+              .then(function() {
+                SUtils.sDebug(`${stage} - ${region} - ${ruleName}: created target "${functionName}:${stage}"`);
+              });
           }
-          return _this.aws.request('CloudWatchEvents', 'putTargets', params, _this.evt.options.stage, _this.evt.options.region);
+          else {
+            SUtils.sDebug(`Removing lambda ${lambdaArn}:${stage} as target for rule ${ruleName} for lambda ${functionName}`);
+
+            let params = {
+              Rule: ruleName,
+              Ids: [ functionName + '_' + stage ]
+            };
+            return _this.aws.request('CloudWatchEvents', 'removeTargets', params, stage, region)
+              .then(function() {
+                 SUtils.sDebug(`${stage} - ${region} - ${ruleName}: removed existing target "${functionName}:${stage}"`);
+              })
+              .catch(function(error) {})
+          }
+
```

Comment:
After thinking about this, this won't be as easy as it sounds. As long as you have an event source defined in `s-function.json`, I think we should also create the respective rule. We _should_ remove it, however, when it's no longer listed in the definition but then it's already too late. At that point we don't know that it ever existed anymore.